core/schema/container.go

```diff
@@ -1113,6 +1121,22 @@ func (s *containerSchema) withFiles(ctx context.Context, parent *core.Container,
 	return parent.WithFiles(ctx, args.Path, files, args.Permissions, args.Owner)
 }
 
+type containerWithoutDirectoryArgs struct {
+	Path string
+}
+
+func (s *containerSchema) withoutDirectory(ctx context.Context, parent *core.Container, args containerWithoutDirectoryArgs) (*core.Container, error) {
+	return parent.WithoutPath(ctx, args.Path)
+}
+
+type containerWithoutFileArgs struct {
+	Path string
+}
+
+func (s *containerSchema) withoutFile(ctx context.Context, parent *core.Container, args containerWithoutFileArgs) (*core.Container, error) {
+	return parent.WithoutPath(ctx, args.Path)
+}
+
```

Comment:
They are actually interchangeable which may seem odd, but it's consistent with `Directory`:

https://github.com/dagger/dagger/blob/aa90870c47d94188969b2c8f4c5b242aa1b08583/core/schema/directory.go#L234-L248

We may want to validate but never seen anyone complain about or notice it in `Directory`.